### PR TITLE
RFC: Creating SIG Models

### PR DESCRIPTION
**Pull request by jaeyounkim**
This RFC will be open for comment until Wednesday, November 11st, 2020.

# Creating SIG Models

We propose to create a TensorFlow SIG Models.

| Status        | Proposed                                             |
:-------------- |:---------------------------------------------------- |
| **RFC #**     |           314                                           |
| **Author(s)** | Jaeyoun Kim (jaeyounkim@google.com), Jing Li (jingli@google.com), Mike Liang (mliang@google.com), Shuangfeng Li (shuangfeng@google.com) |
| **Sponsor**   | Thea Lamkin (thealamkin@google.com)                  |
| **Updated**   | 2020-10-23                                           |

## What is this group for?

This group is for discussions and collaborations on enabling community
contributions to [TensorFlow Model Garden](https://github.com/tensorflow/models)
and [Tensorflow Hub](https://github.com/tensorflow/hub).

SIG Models will focus on empowering the community to contribute state-of-the-art
model implementation in TensorFlow 2. It will benefit the whole community by
providing recommended implementations and models with reproducible results.

SIG Models will have several subgroups (e.g., SIG Models Vision and SIG Models
NLP) covering different machine learning areas. There are several SIG leads for
each group to coordinate the contributions, run community events like contests,
and maintain the code quality through the review process.

### CODE REVIEW

**Comment by sub-mod** [comment]
I would like to be part of the SIG.Thanks

File: rfcs/20201023-sig-models.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+model implementation in TensorFlow 2. It will benefit the whole community by
+providing recommended implementations and models with reproducible results.
+
+SIG Models will have several subgroups (e.g., SIG Models Vision and SIG Models
+NLP) covering different machine learning areas. There are several SIG leads for
+each group to coordinate the contributions, run community events like contests,
+and maintain the code quality through the review process. Each group has the
+flexibility to operate differently.
+
+SIG is also a place for community discussions and sharing best practices of
```
Comment (perfinion) [inline rfcs/20201023-sig-models.md:26]: s/SIG/SIG Models/

File: rfcs/20201023-sig-models.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  submissions
+* Review final submissions and announce results
+
+## What modes of communication do you intend to use?
+
+We plan to have a mailing list
+([models@tensorflow.org](https://groups.google.com/a/tensorflow.org/forum/#!forum/models)),
+arrange a regular (e.g., monthly) video conference call rotated by the SIG
+leads. We will also use a Gitter chat channel
+(https://gitter.im/tensorflow/models) for discussions.
```
Comment (perfinion) [inline rfcs/20201023-sig-models.md:73]: This should be gitter.im/tensorflow/sig-models for consistency with the other sig channels, the existing ones are eg gitter.im/tensorflow/sig-build or gitter.im/tensorflow/sig-addons

**Review by perfinion — COMMENTED** [review]
One thing that is missing from this RFC is a set of guidelines about quality of models in the repo.
SIG-Addons has a good workflow to use as reference. There needs to be a clear statement on quality of the models in the repo and keeping up with maintenance and evolving best-practices. A big risk for a SIG like this is drive-by models that bitrot over time. There must be a process to remove models if they are not up to the standards.

File: rfcs/20201023-sig-models.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+run a contest to motivate the community to contribute state-of-the-art computer
+vision models using TensorFlow 2. 
+
+For example:
+
+* Define/release model implementation tasks, rules, and evaluation criteria for
+  the contest
+* Review initial submissions and provide more support for selected high-quality
+  submissions
+* Review final submissions and announce results
```
Comment (lc0) [inline rfcs/20201023-sig-models.md:65]: what is the lifecycle of such models? I can see at least: draft, initial, final
Any guarantees of maintainability? Can models be deprecated over time?

File: rfcs/20201023-sig-models.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+## What modes of communication do you intend to use?
```
Comment (lc0) [inline rfcs/20201023-sig-models.md:67]: I think would be helpful to also define what is expected use-case.

Do we see it as python module, that can be imported downstream, hence models are tested and maintained.
Or just a collection of scripts, that contributed by different people at different times?

File: rfcs/20201023-sig-models.md (hunk 3 above)
Comment (bhack) [reply rfcs/20201023-sig-models.md:65]: We have already something about this in SIG-addons in the approved RFC https://github.com/tensorflow/community/blob/master/rfcs/20190308-addons-proxy-maintainership.md#repository-growth-and-review /cc @seanpmorgan

File: rfcs/20201023-sig-models.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+NLP) covering different machine learning areas. There are several SIG leads for
+each group to coordinate the contributions, run community events like contests,
+and maintain the code quality through the review process. Each group has the
+flexibility to operate differently.
+
+SIG is also a place for community discussions and sharing best practices of
+using TensorFlow 2 for state-of-the-art research. Furthermore, SIG provides
+product feedback to help TensorFlow to be improved.
+
+## Who will be part of it?
```
Comment (bhack) [inline rfcs/20201023-sig-models.md:30]: What about `keras-cv/keras-nlp`?

File: rfcs/20201023-sig-models.md (hunk 3 above)
Comment (bhack) [reply rfcs/20201023-sig-models.md:65]: We are also trying to figure if a bot could do some keep-alive/MIA checks https://github.com/tensorflow/addons/pull/2024

Comment (bhack) [reply rfcs/20201023-sig-models.md:65]: The real question is also how to implement something like https://github.com/tensorflow/addons/issues/236. I don't know if this new SIG will have the resource to implement something like to solve this open topic. I think it would be really useful for all the SIGs but also for the TF ecosystem.

File: rfcs/20201023-sig-models.md (hunk 4 above)
Comment (bhack) [reply rfcs/20201023-sig-models.md:67]: This comment is connected to the models's reusable components topic.
Some times ago we had a ticket about this at https://github.com/tensorflow/community/issues/223

Comment (bhack) [reply rfcs/20201023-sig-models.md:67]: There are many dups or semi-dups like https://github.com/tensorflow/addons/issues/1903#issuecomment-715897109 that confuse users and third party SIG maintainers.

**Comment by bhack** [comment]
In general I think that is great and strategic to have this SIG cause Models and Hub are really one of the unique places in the ecosystem where all of the pieces converges together tocomposed our end2end goal. Also we often suffer more then the Pytorch ecosystem about availability of **third party papers** official (and unofficial) reference implementations in the github "universe". 

But we really need to take this SIG as an occasion to improve our ecosystem about component re-usability and ownership disambiguation.
It could be nice to have a short grace period for the new components (layer, loss, optimizer, preprocessing, postprocessing, etc.) that would be introduced in TF model.
As we know external members don't know the public roadmap of the different internal teams (and this could happen also between different internal teams) you can find a recap thread at https://github.com/tensorflow/community/issues/29.
This could create duplicates, unaware supersedes, implementation divergences and so on.. In Addons we have tried to Draft an early [ecosystem-review process](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#requirements-for-new-contributions-to-the-repository) to notify the ecosystem that probably could be standardized and improved for general use to improve the coherence between internal teams and external members. 
By this point of view I see this SIG one of the most critical points to achieve this goal in our ecosystem.

File: rfcs/20201023-sig-models.md (hunk 5 above)
Comment (AakashKumarNain) [reply rfcs/20201023-sig-models.md:30]: Same question.

File: rfcs/20201023-sig-models.md (hunk 1 above)
Comment (jaeyounkim) [reply rfcs/20201023-sig-models.md]: @perfinion I've changed it to SIG Models. Thank you!

File: rfcs/20201023-sig-models.md (hunk 2 above)
Comment (jaeyounkim) [reply rfcs/20201023-sig-models.md]: @perfinion I've changed it. Thank you for your comment.

File: rfcs/20201023-sig-models.md (hunk 3 above)
Comment (jaeyounkim) [reply rfcs/20201023-sig-models.md:65]: @lc0 The lifecycle of community models will be one of the topics the SIG Models will discuss. We expect that we will also have a periodic review for models. Thank you for your comment.

@bhack Thank you for your suggestion. I agree with you that it would be useful to have such an analyzer. We will definitely discuss it once the TF SIG Models is created. We will ask for more inputs from you.

File: rfcs/20201023-sig-models.md (hunk 4 above)
Comment (jaeyounkim) [reply rfcs/20201023-sig-models.md:67]: @lc0 @bhank Thank you for your suggestions and comments. We will add that as one of the first topics for open discussion after the SIG Models created. We want to start with a small scope and expand it later based on community inputs.

File: rfcs/20201023-sig-models.md (hunk 5 above)
Comment (jaeyounkim) [reply rfcs/20201023-sig-models.md:30]: @bhack @AakashKumarNain Yes, the Keras team (keras-cv/nlp) will contribute to the SIG Models.

**Review by theadactyl — APPROVED** [review]
lgtm
